packages/musap-rn-kms/src/agent/MusapKeyManagerSystem.ts

```diff
@@ -0,0 +1,127 @@
+    try {
+      const keysJson: MusapKey[] = (await this.musapKeyStore.listKeys()) as MusapKey[];
+      return keysJson.map((key) => this.asMusapKeyInfo(key));
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  async createKey(args: { type: TKeyType; sscdType?: SscdType }): Promise<ManagedKeyInfo> {
+    const keyAlgorithm = this.mapKeyTypeToAlgorithmType(args.type);
+
+    const keyGenReq: KeyGenReq = {
+      keyAlgorithm: keyAlgorithm,
+      did: '',
+      keyUsage: 'sign',
+      keyAlias: uuid(),
+      attributes: [
+        { name: 'purpose', value: 'encrypt' },
+        { name: 'purpose', value: 'decrypt' }
+      ],
+      role: 'administrator'
+    };
+
+    try {
+      const generatedKeyUri = await this.musapKeyStore.generateKey(this.sscdType, keyGenReq);
+      if (generatedKeyUri) {
+        debug('Generated key:', generatedKeyUri);
+        const key = await this.musapKeyStore.getKeyByUri(generatedKeyUri);
+        return this.asMusapKeyInfo(key);
+      } else {
+        throw new Error('Failed to generate key');
+      }
+    } catch (error) {
+      console.error('An error occurred:', error);
+      throw error;
+    }
+  }
+
+  mapKeyTypeToAlgorithmType = (type: TKeyType): KeyAlgorithmType => {
+    switch (type) {
+      case 'Secp256k1':
+        return 'ECCP256K1';
+      case 'Secp256r1':
+        return 'ECCP256R1';
+      case 'RSA':
+        return 'RSA2K';
+      default:
+        throw new Error(`Key type ${type} is not supported by MUSAP`);
+    }
+  }
+
+  async deleteKey({ kid }: { kid: string }): Promise<boolean> {
+    try {
+      await this.musapKeyStore.removeKey(kid);
+      return true;
+    } catch (error) {
+      console.warn('Failed to delete key:', error);
+      return false;
+    }
+  }
+
+  async sign(args: { keyRef: Pick<IKey, 'kid'>; algorithm?: string; data: Uint8Array; [x: string]: any }): Promise<string> {
+      if (!args.keyRef) {
+        throw new Error('key_not_found: No key ref provided');
+      }
+
+      const data = new TextDecoder().decode(args.data as Uint8Array)
+
+      const key: MusapKey = (this.musapKeyStore.getKeyById(args.keyRef.kid)) as MusapKey;
+      const signatureReq: SignatureReq = {
+        keyUri: key.keyUri,
+        data,
+        algorithm: args.algorithm as SignatureAlgorithmType ?? signatureAlgorithmFromKeyAlgorithm(key.algorithm),
+        displayText: args.displayText,
+        transId: args.transId,
+        format: args.format as SignatureFormat ?? 'RAW',
+        attributes: args.attributes
+      }
+      return this.musapKeyStore.sign(signatureReq)
+  }
+
+  async importKey(args: Omit<MinimalImportableKey, 'kms'> & { privateKeyPEM?: string }): Promise<ManagedKeyInfo> {
+    throw new Error('Not supported; MUSAP is a hardware key-store which cannot import keys.');
+  }
+
+  private asMusapKeyInfo(args: MusapKey): ManagedKeyInfo & { keyUri?: string } {
+    return {
+      kid: args.keyId,
+      kms: args.sscdId,
```

Comment:
Why is the kms the sscId name? Other kms-es we can name whatever we want

Reply:
I'm not sure how to answer this. I'm populating the value of `kms` here with the `sscId` that I'm receiving from musap. there are not many options for me in the `MusapKey`:
```
    keyUri: string
    keyAlias: string
    keyType: string
    keyId: string
    sscdId: string
    sscdType: string
    createdDate: string | number // ISO date string
    publicKey: PublicKey
    certificate: MusapCertificate
    certificateChain: MusapCertificate[]
    attributes: KeyAttribute[]
    keyUsages: string[]
    loa: MusapLoA[]
    algorithm: KeyAlgorithm
    did: string
    state: string
    attestation: KeyAttestation
```

Reply:
Why would you store the kms on the key side to begin with. The kms is responsible for knowing what it's name is and what sscd is associated with it, not the other way around

Reply:
Also seeing the LOAs and other options we for sure will need in sprind, I would expect these options to be exposed during key creation

Reply:
ok. will remove populating it

Reply:
also exposed the other values (like the one you mentioned in the meta field of the key)